stonesoup/reader/video.py

```diff
@@ -0,0 +1,220 @@
+# -*- coding: utf-8 -*-
+"""Video readers for Stone Soup.
+
+This is a collection of video readers for Stone Soup, allowing quick reading
+of video data/streams.
+"""
+
+from abc import abstractmethod
+import datetime
+import numpy as np
+import ffmpeg
+import moviepy.editor as mpy
+import threading
+from queue import Queue
+
+from ..base import Property
+from ..buffered_generator import BufferedGenerator
+from ..types.sensordata import ImageFrame
+from .base import SensorDataReader
+from .file import FileReader
+from .url import UrlReader
+
+
+class FrameReader(SensorDataReader):
+    """FrameReader base class
+
+    A FrameReader produces :class:`~.SensorData` in the form of
+    :class:`~ImageFrame` objects.
+    """
+
+    @property
+    def frame(self):
+        return self.sensor_data
+
+    @abstractmethod
+    @BufferedGenerator.generator_method
+    def frames_gen(self):
+        """Returns a generator of frames for each time step.
+
+        Yields
+        ------
+        : :class:`datetime.datetime`
+            Datetime of current time step
+        : set of :class:`~.ImageFrame`
+            Generated frame in the time step
+        """
+        raise NotImplementedError
+
+    @BufferedGenerator.generator_method
+    def sensor_data_gen(self):
+        """Returns a generator of frames for each time step.
+
+        Note
+        ----
+        This is just a wrapper around (and therefore performs identically
+        to) :py:meth:`~frames_gen`.
+
+        Yields
+        ------
+        : :class:`datetime.datetime`
+            Datetime of current time step
+        : set of :class:`~.ImageFrame`
+            Generated frame in the time step
+        """
+        return self.frames_gen()
```

Comment:
I think this should be `yield from self.frames_gen()`.

Reply:
Done